npm/private/npm_package_store.bzl

```diff
@@ -205,25 +205,20 @@ def _npm_package_store_impl(ctx):
         else:
             package_store_directory = ctx.actions.declare_directory(package_store_directory_path)
             if utils.is_tarball_extension(src.extension):
-                # npm packages are always published with one top-level directory inside the tarball, tho the name is not predictable
-                # we can use the --strip-components 1 argument with tar to strip one directory level
-                args = ctx.actions.args()
-                args.add("--extract")
-                args.add("--no-same-owner")
-                args.add("--no-same-permissions")
-                args.add("--strip-components")
-                args.add(str(1))
-                args.add("--file")
-                args.add(src.path)
-                args.add("--directory")
-                args.add(package_store_directory.path)
-
+                # npm packages are always published with one top-level directory inside the tarball,
+                # tho the name is not predictable we can use the --strip-components 1 argument with
+                # tar to strip one directory level. Some packages have directory permissions missing
+                # executable which make the directories not listable (pngjs@5.0.0 for example). Run
+                # `chmod -R a+X` to fix up these packages (https://stackoverflow.com/a/14634721).
+                # See https://github.com/aspect-build/rules_js/issues/1637 for more info.
                 bsdtar = ctx.toolchains["@aspect_bazel_lib//lib:tar_toolchain_type"]
-                ctx.actions.run(
-                    executable = bsdtar.tarinfo.binary,
+                ctx.actions.run_shell(
+                    tools = [bsdtar.tarinfo.binary],
                     inputs = depset(direct = [src], transitive = [bsdtar.default.files]),
                     outputs = [package_store_directory],
-                    arguments = [args],
```

Comment:
Why can't we still use `arguments`?

Reply:
There is an && in the command which makes args confusing since there are two commands here

Reply:
Is there any risk in manually concatenating args? Is it worth fixing a rare edge case for that?

Would it be better patching the bad package instead? etc...

Reply:
I'd also be curious if this can be fixed with a tar command instead?

Reply:
No way that I know of. We ran into the same issue in npm_import.bzl after extracting packages in the repo rule.

Reply:
I don't think putting args in to the `command` is an issue. Actually, reading the docstring on arguments for run_shell, you're still required to set `$1` `$2` in the command to reference them either way: https://bazel.build/rules/lib/builtins/actions#run_shell

Reply:
Yea. That works nicely.

Reply:
Yes that's much better imo.

`ctx.action.args()` should be used whenever possible though. In this case when `command` is just a string (no concatenation, no depset `.to_list` etc) it might be fine. See https://bazel.build/rules/performance#ctx-actions-args